File: go-backend-auth/models/wilayahModel.go
package models

import "time"

type Wilayah struct {
	// Kita custom Primary Key supaya sesuai dengan SQL kamu (id_wilayah)
	ID        uint      `gorm:"primaryKey;column:id_wilayah" json:"id_wilayah"`
	Kabupaten string    `gorm:"type:varchar(100);not null" json:"kabupaten"`
	Kecamatan string    `gorm:"type:varchar(100);not null" json:"kecamatan"`
	Latitude  float64   `gorm:"type:decimal(10,6)" json:"latitude"`
	Longitude float64   `gorm:"type:decimal(10,6)" json:"longitude"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

// Override nama tabel agar GORM menggunakan 'wilayah' (bukan wilayahs)
func (Wilayah) TableName() string {
	return "wilayah"
}

File: lib/features/admin/main_data/units/data/models/wilayah_model.dart
class WilayahModel {
  final int id;
  final String kabupaten;
  final String kecamatan;
  final double latitude;
  final double longitude;

  WilayahModel({
    required this.id,
    required this.kabupaten,
    required this.kecamatan,
    required this.latitude,
    required this.longitude,
  });

  factory WilayahModel.fromJson(Map<String, dynamic> json) {
    return WilayahModel(
      id: json['id_wilayah'] ?? 0,
      kabupaten: json['kabupaten'] ?? '',
      kecamatan: json['kecamatan'] ?? '',
      latitude: (json['latitude'] ?? 0.0).toDouble(),
      longitude: (json['longitude'] ?? 0.0).toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id_wilayah': id,
      'kabupaten': kabupaten,
      'kecamatan': kecamatan,
      'latitude': latitude,
      'longitude': longitude,
    };
  }
}